src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyPlainTextCommentFilter.java

```diff
@@ -0,0 +1,669 @@
+ * static final int lowerCaseConstant2;
+ * static final int lowerCaseConstant3;
+ * static final int lowerCaseConstant4;
+ * static final int lowerCaseConstant5; // violation
+ * </pre>
+ * <p>
+ * To configure the filter to suppress audit events for the current and previous 2 lines:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;Checker&quot;&gt;
+ *   &lt;module name=&quot;SuppressWithNearbyPlainTextCommentFilter&quot;&gt;
+ *     &lt;property name=&quot;influenceFormat&quot; value=&quot;-2&quot;/&gt;
+ *   &lt;/module&gt;
+ *   &lt;module name=&quot;TreeWalker&quot;&gt;
+ *     &lt;module name=&quot;ConstantNameCheck&quot;/&gt;
+ *   &lt;/module&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * static final int lowerCaseConstant; // violation
+ * static final int lowerCaseConstant1;
+ * static final int lowerCaseConstant2;
+ * static final int lowerCaseConstant3; // SUPPRESS CHECKSTYLE the reason for this is..
+ * static final int lowerCaseConstant4; // violation
+ * static final int lowerCaseConstant5; // violation
+ * </pre>
+ * <p>
+ * To configure the filter to accept variable {@code checkFormat} and {@code influenceFormat}:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;Checker&quot;&gt;
+ *   &lt;module name=&quot;SuppressWithNearbyPlainTextCommentFilter&quot;&gt;
+ *     &lt;property name=&quot;commentFormat&quot;
+ *       value=&quot;SUPPRESS CHECKSTYLE (\w+) - ([+-]\d+) Lines&quot;/&gt;
+ *     &lt;property name=&quot;commentFormat&quot; value=&quot;$1&quot;/&gt;
+ *     &lt;property name=&quot;influenceFormat&quot; value=&quot;$2&quot;/&gt;
+ *   &lt;/module&gt;
+ *   &lt;module name=&quot;TreeWalker&quot;&gt;
+ *     &lt;module name=&quot;ConstantNameCheck&quot;/&gt;
+ *   &lt;/module&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * static final int lowerCaseConstant; // violation
+ * static final int lowerCaseConstant1; // SUPPRESS CHECKSTYLE FileTabCharacterCheck - +3 Lines
+ * static final int lowerCaseConstant2;
+ * static final int lowerCaseConstant3;
+ * static final int lowerCaseConstant4;
+ * static final int lowerCaseConstant5; // violation
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.Checker}
+ * </p>
+ *
+ * @since 10.4
+ */
+public class SuppressWithNearbyPlainTextCommentFilter
+    extends AutomaticBean
+    implements Filter {
+
+    /** Default comment format to turn check reporting off. */
+    private static final String DEFAULT_COMMENT_FORMAT = "SUPPRESS CHECKSTYLE (\\w+)";
+
+    /** Default regex for checks that should be suppressed. */
+    private static final String DEFAULT_CHECK_FORMAT = ".*";
+
+    /** Default number of lines that should be suppressed. */
+    private static final String DEFAULT_INFLUENCE_FORMAT = "0";
+
+    /** Stores all suppressions for a given file name. */
+    private final Map<String, List<Suppression>> fileSuppressions = new HashMap<>();
```

Comment:
Uh, so the first 4 surviving mutations are from the hashmap I added to improve performance. I can easily remove it(just like `SuppressWithPlainTextCommentFilter` works) but there are so many redundant calls to `getSuppressions()` that I wish could be avoided.
```
Source File: "SuppressWithNearbyPlainTextCommentFilter.java"
Class: "com.puppycrawl.tools.checkstyle.filters.SuppressWithNearbyPlainTextCommentFilter"
Method: "getSuppressions"
Line Contents: "fileSuppressions.putIfAbsent(fullFileName, suppressions);"
Mutator: "org.pitest.mutationtest.engine.gregor.mutators.NonVoidMethodCallMutator"
Description: "removed call to java/util/Map::putIfAbsent"
Line Number: 437

Source File: "SuppressWithNearbyPlainTextCommentFilter.java"
Class: "com.puppycrawl.tools.checkstyle.filters.SuppressWithNearbyPlainTextCommentFilter"
Method: "getSuppressions"
Line Contents: "fileSuppressions.putIfAbsent(fullFileName, suppressions);"
Mutator: "org.pitest.mutationtest.engine.gregor.mutators.experimental.ArgumentPropagationMutator"
Description: "replaced call to java/util/Map::putIfAbsent with argument"
Line Number: 437

Source File: "SuppressWithNearbyPlainTextCommentFilter.java"
Class: "com.puppycrawl.tools.checkstyle.filters.SuppressWithNearbyPlainTextCommentFilter"
Method: "getSuppressions"
Line Contents: "final String fullFileName = fileText.getFile().getPath();"
Mutator: "org.pitest.mutationtest.engine.gregor.mutators.NonVoidMethodCallMutator"
Description: "removed call to java/io/File::getPath"
Line Number: 428

Source File: "SuppressWithNearbyPlainTextCommentFilter.java"
Class: "com.puppycrawl.tools.checkstyle.filters.SuppressWithNearbyPlainTextCommentFilter"
Method: "getSuppressions"
Line Contents: "if (suppressions.isEmpty()) {"
Mutator: "org.pitest.mutationtest.engine.gregor.mutators.RemoveConditionalMutator_EQUAL_IF"
Description: "removed conditional - replaced equality check with true"
Line Number: 432
```

Reply:
Let's focus on functionality/ no mutations first, then work on optimizations :)

Reply:
Alright. Shall I remove the hashmap so the mutations go away and later we can talk about putting it back?